src/util/irep.h

```diff
@@ -21,6 +21,7 @@ Author: Daniel Kroening, kroening@kroening.com
 // #define HASH_CODE
 #define USE_MOVE
 // #define NAMED_SUB_IS_FORWARD_LIST
+#define IS_CANONICAL_BIT
```

Comment:
I have feelings on this subject and I feel compelled to share them with you.

1. Before this can be merged we *need* to add the assertion to the simplifier that checks idempotency.  By design it should hold but ... I've been known to bet people up to 50p that it doesn't in one or more obscure cases.

2. We should also think about the cache in the simplifier.

3. Crazy idea : rather than just a bit, we could make simplification just a bit more union-find and have a pointer to the canonical version.  This would allow fast simplification and deduplification / re-introduction of sharing.

4. This road starts to feel like it leads to canonical irept's.  I remember discussing this ages ago and coming to the conclusion that this was common in SMT solvers because their nodes were simple and unchanging while CPROVER really needed them to be able to gain annotations / fields.  If so this could be ... a thing.